src/olympia/addons/tasks.py

```diff
@@ -536,28 +536,13 @@ def flag_high_hotness_according_to_review_tier():
         # Need a hotness threshold to be set for the tier.
         growth_threshold_before_flagging__isnull=False,
     )
-    base_qs = Addon.unfiltered.exclude(status=amo.STATUS_DISABLED).filter(
-        type=amo.ADDON_EXTENSION
-    )
+    # Build a single queryset with all add-ons we want to flag for each tier.
```

Comment:
It would be a lot easier to review this if we had 2 commits. 1) migrate the code to the new structure moving the logic to the model and then 2) modify the logic calculating the threshold filters.

Is that too big an ask? I feel I'm spending a lot of time trying to map against the two types of changes.

Reply:
I actually did that but rebased it away :sweat_smile: Ultimately the algorithm change is extremely straightforward, since it's just a single line in `get_growth_threshold()`.

Reply:
See my other comment: https://github.com/mozilla/addons-server/pull/22875/files#r1853839986